cfme/configure/access_control.py

```diff
@@ -332,3 +360,98 @@ def copy(self, name=None):
              action=form_buttons.add)
         flash.assert_success_message('Role "%s" was saved' % new_role.name)
         return new_role
+
+
+class Tenant(Updateable, Pretty):
+    """Kudos to mfalesni"""
+
+    save_changes = form_buttons.FormButton("Save changes")
+    tenant_form = Form(
+        fields=[
+            ('name', Input('name')),
+            ('description', Input('description'))
+        ])
+    pretty_attrs = ["name", "description"]
+
+    @classmethod
+    def get_root_tenant(cls):
+        return cls(name="My Company", _default=True)
+
+    def __init__(self, name=None, description=None, parent_tenant=None, _default=False):
+        self.name = name
+        self.description = description
+        self.parent_tenant = parent_tenant
+        self._default = _default
+
+    @property
+    def parent_tenant(self):
+        if self._default:
+            return None
+        if self._parent_tenant:
+            return self._parent_tenant
+        return self.get_root_tenant()
+
+    @parent_tenant.setter
+    def parent_tenant(self, tenant):
+        if tenant is not None and isinstance(tenant, Project):
+            # If we try to
+            raise ValueError("Project cannot be a parent object.")
+        if isinstance(tenant, basestring):
+            # If parent tenant is passed as string, we assume that tenant name was passed instead of object
+            tenant = Tenant(tenant)
+        self._parent_tenant = tenant
```

Comment:
I think this:
```python
if tenant == self:
    raise ValueError("Setting parent tenant as the same tenant would cause infinite recursion")
```

But I don't know whether you can have 2 tenants of the same name although in different place in the tree. If yes, then `__eq__` would have to be modified to recursively compare also parents. If not, then it should work properly.